lib/bamboo/adapters/local_adapter.ex

```diff
@@ -32,18 +32,8 @@ defmodule Bamboo.LocalAdapter do
     case config[:deliver_later_strategy] do
       nil ->
         Map.put(config, :deliver_later_strategy, Bamboo.DeliverImmediatelyStrategy)
```

Comment:
I actually think we can just return the config as-is and use `TaskSupervisorStrategy` that is set by default in the mailer: https://github.com/paulcsmith/bamboo/blob/master/lib/bamboo/mailer.ex#L145